spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala

```diff
@@ -73,6 +73,47 @@ abstract class AbstractModule[A <: Activity: ClassTag, B <: Activity: ClassTag,
    */
   var gradInput: A = Activity.allocate[A, T]()
 
+
+  private var inputShapeValue: Activity = null
+
+  private var outputShapeValue: Activity = null
+
+  def setBatchInputShape(inputShape: Activity): Unit = {
+    this.inputShapeValue = inputShape
+  }
+
+  def setBatchOutputShape(outputShape: Activity): Unit = {
+    this.outputShapeValue = outputShape
+  }
+
+
+  def getBatchInputShape(): Activity = {
+    inputShapeValue
+  }
+
+  def getBatchOutputShape(): Activity = {
```

Comment:
what if user change the input shape after call build? does the output shape information correct?